### ISSUE TITLE
Debug output of `SMatrix` is confusing

### ISSUE THREAD

**Issue by Andlon** [dimforge/nalgebra#898]
Consider the following code:

```rust
let matrix = Matrix2::new(1.0, 2.0, 3.0, 4.0);
println!("{:?}", matrix);
```
The output is:
```text
Matrix { data: [[1.0, 3.0], [2.0, 4.0]] }
```
As a casual user, just looking at the `Debug` output I'm likely to think that the matrix stored here is
```
[ 1   3 ]
[ 2   4 ]
```
whereas this is the transposed of the true matrix. The reason here is of course that `nalgebra` stores its matrices in a column-major format. However, perhaps we do not need to expose this fact in the `Debug` implementation. An idea here could be to use the same format as our new `matrix!` macro. In that case, the one-line output of `Debug` for the above matrix would read:
```text
[1.0, 2.0; 3.0, 4.0]
```
whereas with a pretty-print specifier (`"{:#?}"`) we could print it formatted with line breaks:
```text
[1.0, 2.0;
 3.0, 4.0]
```

This issue led to some considerable confusion lately in #893 as a user wanted to port some code from `numpy`, and looking at the debug output it appeared that the output matrices were transposed.

**Comment by iMplode-nZ** [comment]
I would also like this, the format is really confusing right now.

**Comment by Ralith** [comment]
I'd love a more concise format than the current one; reading the `Debug` output of a large structure, or large collection of structures, containing nalgebra data is presently a chore.

**Comment by Andlon** [comment]
@iMplode-nZ, @Ralith: any thoughts on the proposal of using the `matrix!` macro syntax? It has the benefit that the matrix data can be directly copy-paste into code for inspection (for example you can print it from application code and paste it into a unit test or similar).

**Comment by iMplode-nZ** [comment]
No, just print it `Matrix[[a b c]]` (with special cases for vectors) as using `matrix!` is confusing.
It'd be like printing vectors as `vec![]` instead of just `[]`.

**Comment by OfficialURL** [comment]
> No, just print it `Matrix[[a b c]]` (with special cases for vectors) as using `matrix!` is confusing.
> It'd be like printing vectors as `vec![]` instead of just `[]`.

I don't think that was the proposal. I thought the proposal was to write down matrices in a way similar as you do in the `matrix!` macro (i.e. `[1, 2; 3, 4]`) but without actually writing down `matrix!` in the output. That I agree with.

**Comment by Andlon** [comment]
Yes, sorry if this was not so clear. What I intended is as @OfficialURL suggested.

**Comment by iMplode-nZ** [comment]
That seems fine, although with the `{:#?}` it should format the thing as well. Also make sure to tag the thing as `Matrix` so that its clear it isn't a `Vec`
